### PR TITLE
Address insensitive language

### PR DESCRIPTION
**Pull request by nrb**
This change will update all links pointing to the `master` branch to instead point to a `main` branch in the Velero codebase.

There are still references to `master` and `blacklist` within the code, based on 3rd party dependencies.

The links to `main` files will be broken until we actually make the switch from `master` to `main` on GitHub; I just want to verify it works now.

### CODE REVIEW

File: site/_posts/2019-08-22-announcing-velero-1.1.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ We’ve also made several other enhancements to Velero that should benefit all u
...
 
-We’ve also taken some time to improve Velero for the future by updating the Deployment and DaemonSet to use the apps/v1 API group, which will be the [default in Kubernetes 1.16](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.16.md#action-required-3). This change means that `velero install` and the `velero plugin` commands will require Kubernetes 1.9 or later to work. Existing Velero installs will continue to work without needing changes, however.
+We’ve also taken some time to improve Velero for the future by updating the Deployment and DaemonSet to use the apps/v1 API group, which will be the [default in Kubernetes 1.16](https://github.com/kubernetes/kubernetes/blob/main/CHANGELOG-1.16.md#action-required-3). This change means that `velero install` and the `velero plugin` commands will require Kubernetes 1.9 or later to work. Existing Velero installs will continue to work without needing changes, however.
```
Comment (ashish-amarnath) [inline site/_posts/2019-08-22-announcing-velero-1.1.md:39]: https://github.com/kubernetes/kubernetes/blob/main/CHANGELOG-1.16.md#action-required-3 is broken

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Master CI
+name: Main CI
```
Comment (nrb) [inline .github/workflows/push.yml:1]: Until this is merged, this is going to be a broken badge on previews, I think.

File: site/_posts/2019-10-10-Velero-v1-1-Stateful-Backup-vSphere.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ Velero version 1.1 provides support to backup applications orchestrated on upstr
...
 
-For instructions on how to download and deploy a simple Cassandra StatefulSet, please refer to [this blog post](https://cormachogan.com/2019/06/12/kubernetes-storage-on-vsphere-101-statefulset/). This will show you how to deploy a Cassandra StatefulSet which we can use to do our Stateful aplication backup and restore. The manifests [available here](https://github.com/cormachogan/vsphere-storage-101/tree/master/StatefulSets) use an earlier version of Cassandra (v11) that includes the `cqlsh` tool which we will use now to create a database and populate a table with some sample data.
+For instructions on how to download and deploy a simple Cassandra StatefulSet, please refer to [this blog post](https://cormachogan.com/2019/06/12/kubernetes-storage-on-vsphere-101-statefulset/). This will show you how to deploy a Cassandra StatefulSet which we can use to do our Stateful aplication backup and restore. The manifests [available here](https://github.com/cormachogan/vsphere-storage-101/tree/main/StatefulSets) use an earlier version of Cassandra (v11) that includes the `cqlsh` tool which we will use now to create a database and populate a table with some sample data.
```
Comment (ashish-amarnath) [inline site/_posts/2019-10-10-Velero-v1-1-Stateful-Backup-vSphere.md:32]: broken link

File: site/_posts/2020-05-27-CSI-integration.md [hunk 4]
```diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
 
 This post uses an AKS cluster running Kubernetes `1.17`, with Azure disk CSI driver as an example.
 
-Following instructions to install the Azure disk CSI driver from [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/docs/install-csi-driver-master.md) run the below command
+Following instructions to install the Azure disk CSI driver from [here](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/docs/install-csi-driver-main.md) run the below command
```
Comment (ashish-amarnath) [inline site/_posts/2020-05-27-CSI-integration.md:30]: broken link

File: site/_posts/2020-05-27-CSI-integration.md [hunk 5]
```diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
...
 
 ```bash 
-curl -skSL https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/master/deploy/install-driver.sh | bash -s master snapshot -- 
+curl -skSL https://raw.githubusercontent.com/kubernetes-sigs/azuredisk-csi-driver/main/deploy/install-driver.sh | bash -s main snapshot --
```
Comment (ashish-amarnath) [inline site/_posts/2020-05-27-CSI-integration.md:33]: the link and the bash sub-command are not valid

File: site/_posts/2020-05-27-CSI-integration.md [hunk 6]
```diff
@@ -27,17 +27,17 @@ As the [Kubernetes CSI Snapshot Beta API](https://kubernetes.io/docs/concepts/st
...
 ```
 
 This script will deploy the following CSI components, CRDs, and necessary RBAC:
 
-- [`deployment.apps/csi-azuredisk-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-controller.yaml)
-- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-azuredisk-node.yaml)
-- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/csi-snapshot-controller.yaml)
+- [`deployment.apps/csi-azuredisk-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-azuredisk-controller.yaml)
+- [`daemonset.apps/csi-azuredisk-node`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-azuredisk-node.yaml)
+- [`deployment.apps/csi-snapshot-controller`](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/csi-snapshot-controller.yaml)
```
Comment (ashish-amarnath) [inline site/_posts/2020-05-27-CSI-integration.md:40]: broken links

File: site/_posts/2020-05-27-CSI-integration.md [hunk 7]
```diff
@@ -85,7 +85,7 @@ parameters:
   tags: 'foo=aaa,bar=bbb' 
 ```
 
-NOTE: The above `yaml` was sourced from [StorageClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/storageclass-azuredisk-csi.yaml) and [VolumeSnapshotClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/master/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml).
+NOTE: The above `yaml` was sourced from [StorageClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/example/storageclass-azuredisk-csi.yaml) and [VolumeSnapshotClass](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main/deploy/example/snapshot/storageclass-azuredisk-snapshot.yaml).
```
Comment (ashish-amarnath) [inline site/_posts/2020-05-27-CSI-integration.md:88]: broken links

**Comment by nrb** [comment]
@a-mccarthy Thanks!

In terms of the docs site, yeah, I think we should investigate redirecting master to main. In terms of the git branch, I'm not sure how that will work; we can rename our own branches, but I don't think GitHub does redirects for renamed branches, so we'll need to do some research.

Some possible solutions for docs:

 * Get a redirect solution from master to main
 * Leave master docs in place, but remove them from the ToC and put a custom banner in place referencing the name change, and stop updating them.
 * Something else?

Also @tbatard, this will affect #2645, so I want to give you a heads up.

**Comment by a-mccarthy** [comment]
Good point about Github, i was referring only to docs, but i'm not sure that Github does redirects for branches either.

I personally like adding a redirect to the docs site, bc it has less overheard than keeping the master docs folder around and adding a warning. Netlify handles redirects in the [config file](https://docs.netlify.com/routing/redirects/#syntax-for-the-netlify-configuration-file)

**Comment by nrb** [comment]
Renamed docker master tag

**Comment by nrb** [comment]
The redirects for documentation are broken - it looks like the splat directive adds the whole of the path after the domain. Need to figure out how to just redirect the part after `master`/`main`.

File: netlify.toml [hunk 8]
```diff
@@ -2,3 +2,9 @@
   base = "site/"
   command = "jekyll build"
   publish = "site/_site"
+
+# Using the * -> :splat rule should redirect anything under master to the matching file under main
+[[redirects]]
+  from = "docs/master/*"
```
Comment (a-mccarthy) [inline netlify.toml:8]: @nrb i _think_ you might just need to have a / at the beginning of the to/from values. hoping its that straight forward :)

Comment (nrb) [reply netlify.toml]: Thanks, that looks like it worked!

**Comment by nrb** [comment]
@ashish-amarnath @carlisia @a-mccarthy Ok, I think docs are now working and I've updated all of our own scripts, code, and design docs that I'm aware of. I've also rebased it to the latest state of things.

I would appreciate another double check on this, then perhaps we do the switch over on GitHub either tomorrow (July 14) or Thursday (July 15).

Also, @ashish-amarnath or @carlisia, you would both be willing to help with getting the plugins updated, those should be easier, as there's no Netlify docs associated with them. The main replacements I found in here were:

master -> main (only 1 instance of this outside of branch tooling/links)
blacklist -> exclusion or deny list (depending on context)
whitelist -> include or allow list (depending on context)
slave -> secondary or follower (I didn't find any reference to this in Velero's core code, but there were some in the dependencies that we didn't use)

**Comment by carlisia** [comment]
I can take the Helm/Azure/GCP. If @ashish-amarnath if overloaded I can take the others as well.

**Comment by nrb** [comment]
>I can take the Helm/Azure/GCP.

Thanks! If we have any problems with those, we can discuss them in their own repos.

File: design/csi-snapshots.md [hunk 9]
```diff
@@ -306,16 +306,16 @@ Without these objects, the provider-level snapshots cannot be located in order t
...
+[5]: https://github.com/heptio/velero/blob/main/pkg/apis/velero/v1/pod_volume_backup.go#L88
 [6]: https://github.com/heptio/velero-csi-plugin/
-[7]: https://github.com/heptio/velero/blob/master/pkg/plugin/velero/volume_snapshotter.go#L26
-[8]: https://github.com/heptio/velero/blob/master/pkg/controller/backup_controller.go#L560
-[9]: https://github.com/heptio/velero/blob/master/pkg/persistence/object_store.go#L46
-[10]: https://github.com/heptio/velero/blob/master/pkg/apis/velero/v1/labels_annotations.go#L21
-[11]: https://github.com/heptio/velero/blob/master/pkg/cmd/server/server.go#L471
-[12]: https://github.com/heptio/velero/blob/master/pkg/cmd/util/output/backup_describer.go
-[13]: https://github.com/heptio/velero/blob/master/pkg/cmd/util/output/backup_describer.go#L214
+[7]: https://github.com/heptio/velero/blob/main/pkg/plugin/velero/volume_snapshotter.go#L26
```
Comment (ashish-amarnath) [inline design/csi-snapshots.md:312]: Do we want to cleanup the heptio reference here as well?
Not a blocker.

**Comment by carlisia** [comment]
Does this need to be edited to merge against `main`?

**Comment by nrb** [comment]
>Does this need to be edited to merge against main?

@carlisia My intention was to merge into `master` then re-create `main` from that commit. However, I could reverse that order of operations. Recreate `main` so that it's up-to-date, edit the PR to go into `main`, then merge it. That does seem cleaner.

The reason I want to recreate `main` is that it seems cleaner than doing a bunch of merges/rebases since it's a brand new branch.

File: design/csi-snapshots.md (hunk 9 above)
Comment (nrb) [reply design/csi-snapshots.md:312]: Might as well, I tried to get ones I saw, but didn't search for them.

**Comment by nrb** [comment]
@ashish-amarnath - Response inline, monospaced

    https://github.com/nrb/velero/blob/insensitive-language/Makefile#L42 - `Done`
    https://raw.githubusercontent.com/nrb/velero/insensitive-language/CHANGELOG.md - `done`
    Files in https://github.com/nrb/velero/tree/insensitive-language/pkg/backup/testdata/v1 - `I purposefully didn't touch these as they're external CRDs and I wanted to retain their state as-is so we're not changing things.`
    https://github.com/nrb/velero/blob/insensitive-language/site/js/jquery.matchHeight.js#L146 IDK if we should be changing this, but flagging for visibility - `Yeah, here I left it alone cause I have no idea if it'll break anything`.
    Couple of occurrences in https://github.com/nrb/velero/blob/insensitive-language/design/plugin-backup-and-restore-progress-design.md - `Done`
    Some change logs have references to master - `Done, except for links to kubernetes`.

**Comment by carlisia** [comment]
> Recreate main so that it's up-to-date, edit the PR to go into main, then merge it. That does seem cleaner.

@nrb I think this is cleaner too. I'll do the same in the other repos.

**Comment by nrb** [comment]
Target branch updated.

**Pull request merged**